tests/test_routes.py

```diff
@@ -55,21 +73,81 @@ def test_index(self):
+        location = response.headers.get("Location", None)
+        self.assertIsNotNone(location)
+
+        # Check the data is correct
+        new_product = response.get_json()
+        self.assertEqual(new_product["name"], test_product.name)
+        self.assertEqual(new_product["price"], test_product.price)
+        self.assertEqual(new_product["description"], test_product.description)
+        self.assertEqual(new_product["status"], test_product.status.name)
+        self.assertEqual(new_product["url"], test_product.url)
+        self.assertEqual(new_product["rating"], test_product.rating)
+        self.assertEqual(new_product["category"], test_product.category)
+
+        # Check that the location header was correct
+        response = self.client.get(location)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_read_product(self):
+        """It should Get a single Product"""
+        # get the id of a pet
+        test_product = self._create_products(1)[0]
+        response = self.client.get(f"{BASE_URL}/{test_product.id}")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.get_json()
+        self.assertEqual(data["name"], test_product.name)
+        self.assertEqual(data["price"], test_product.price)
+        self.assertEqual(data["description"], test_product.description)
+        self.assertEqual(data["status"], test_product.status.name)
+
+    def test_read_product_not_found(self):
+        """It should not Get a Product thats not found"""
+        response = self.client.get(f"{BASE_URL}/0")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        data = response.get_json()
+        logging.debug("Response data = %s", data)
+        self.assertIn("was not found", data["message"])
+
     # Todo: Add your test cases here...
-    def test_create_product_success(self):
-        response = self.client.post('/products', data=json.dumps(self.product_data), content_type='application/json')
-        self.assertEqual(response.status_code, 201)
-        self.assertIn('Test Product', response.data.decode())
-
-    def test_create_product_already_exists(self):
-        self.client.post('/products', data=json.dumps(self.product_data), content_type='application/json')
-        response = self.client.post('/products', data=json.dumps(self.product_data), content_type='application/json')
-        self.assertEqual(response.status_code, 409)
-        self.assertIn('already exists', response.data.decode())
-
-    def test_create_product_invalid_data(self):
-        invalid_data = self.product_data.copy()
-        del invalid_data['price']
-        response = self.client.post('/products', data=json.dumps(invalid_data), content_type='application/json')
-        self.assertEqual(response.status_code, 400)
-        self.assertIn('Invalid data', response.data.decode())
\ No newline at end of file
+    # def test_create_product_success(self):
```

Comment:
**Nit:** do you think we should delete this commented out code?

Reply:
I think we can let it stay and use it to add additional code coverage